### PR TITLE
refactor: remove ErrorSpan in dependencies

### PR DESCRIPTION
**Pull request by shulaoda**
## Summary
Related to #7422 

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Comment by hardfist** [comment]
we're preparing for the 1.0 release, so hold this pr until 1.0 is released(this wedenesday)

File: crates/rspack_plugin_css/src/dependency/url.rs [hunk 1]
```diff
@@ -75,7 +65,7 @@ impl Dependency for CssUrlDependency {
   }
 
   fn span(&self) -> Option<ErrorSpan> {
```
Comment (h-a-n-a) [inline crates/rspack_plugin_css/src/dependency/url.rs:67]: Do we still have to keep `ErrorSpan`? Wouldn't `RealDependencyLocation` be enough?

Comment (shulaoda) [reply crates/rspack_plugin_css/src/dependency/url.rs:67]: This is a huge historical legacy issue, and I plan to address it in the next few PRs.

Comment (h-a-n-a) [reply crates/rspack_plugin_css/src/dependency/url.rs:67]: Thanks for the clarification.

Comment (h-a-n-a) [reply crates/rspack_plugin_css/src/dependency/url.rs:67]: Would you please rebase and resolve the conflict? I would happy to get this merged.

**Comment by shulaoda** [comment]
Encountered this issue #7683, which is currently frequently encountered in CI/CD.

**Review by h-a-n-a — APPROVED** [review]

**Pull request merged**